src/Storages/MergeTree/MergeTreeWriteAheadLog.cpp

```diff
@@ -234,6 +241,15 @@ MergeTreeData::MutableDataPartsVector MergeTreeWriteAheadLog::restore(const Stor
         }
     }
 
+    /// All parts in WAL had been already committed into the disk -> clear the WAL
+    if (parts.empty())
+    {
+        LOG_DEBUG(log, "WAL file '{}' had been completely processed. Removing.", path);
+        disk->removeFile(path);
+        init();
+        return {};
+    }
+
```

Comment:
Maybe do this check after removing of dropped parts, because looks like for them `getActiveContainingPart` may return `false`, but we don't need to process them?

Reply:
You are right, nice catch! Rebased.